### PR TITLE
add Y12 intrument format

### PR DESCRIPTION
**Pull request by jpcima**
Y12 is an instrument dump from the emulator Gens KMod.
It's also one of the import formats for DefleMask, there are some musicians who request it.
Nothing special about it, it's a dump of registers.
http://gendev.spritesmind.net/forum/viewtopic.php?t=1252

### CODE REVIEW

**Comment by jpcima** [comment]
Samples of Y12
[Streets of Rage (W) (REV01) [!]_Chan1.y12.gz](https://github.com/Wohlstand/OPN2BankEditor/files/2714523/Streets.of.Rage.W.REV01._Chan1.y12.gz)
[Sonic The Hedgehog (W) (REV01) [!]_Chan3.y12.gz](https://github.com/Wohlstand/OPN2BankEditor/files/2714524/Sonic.The.Hedgehog.W.REV01._Chan3.y12.gz)

**Comment by Wohlstand** [comment]
Nice! :fox_face: :+1: 
Will check it out at evening...

**Comment by Wohlstand** [comment]
P.S. Can you put few sample files into "Bank_Samples" folder to allow everyone have samples of each supported format?

File: src/FileFormats/format_gens_y12.cpp [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "format_gens_y12.h"
+#include "../common.h"
+
+#include <QFileInfo>
+
+bool Gens_Y12::detectInst(const QString &filePath, char *magic)
```
Comment (Wohlstand) [inline src/FileFormats/format_gens_y12.cpp]: Please avoid warnings, and keep `char * /*magic*/` if you don't use this field.

**Comment by jpcima** [comment]
>P.S. Can you put few sample files into "Bank_Samples" folder to allow everyone have samples of each supported format?

Yeah I'll do. Next, I'll follow with a second file format which vastly more interesting than this one is.
And has lots of samples :wink:

**Comment by Wohlstand** [comment]
Okay, I'll merge this and will debug result at evening if needed.

**Pull request merged**